prov/gni/src/gnix_av.c

```diff
@@ -36,13 +36,449 @@
 //
 #include <stdlib.h>
 #include <string.h>
+#include <inttypes.h>
 
 #include "gnix.h"
 #include "gnix_util.h"
 
+
+/*******************************************************************************
+ * Forward declarations of ops structures.
+ ******************************************************************************/
+static struct fi_ops_av gnix_av_ops;
+static struct fi_ops gnix_fi_av_ops;
+
+/*******************************************************************************
+ * Helper functions.
+ ******************************************************************************/
+/*
+ * TODO: Check for named AV creation and check RX CTX bits.
+ */
+static int gnix_verify_av_attr(struct fi_av_attr *attr)
+{
+	int ret = FI_SUCCESS;
+
+	switch (attr->type) {
+	case FI_AV_TABLE:
+	case FI_AV_MAP:
+		break;
+	default:
+		ret = -FI_EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+/*
+ * Check the capacity of the internal table used to represent FI_AV_TABLE type
+ * address vectors. Initially the table starts with a capacity and count of 0
+ * and the capacity increases by roughly double each time.
```

Comment:
Yes, sorry that was just a poorly worded comment. It increases in size by 2*x + 1 everytime it gets resized which is only when the internal count + requested additions are greater than the capacity.